clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java

```diff
@@ -2471,6 +2477,8 @@ public void testCurrentLag(GroupProtocol groupProtocol) {
 
         // poll once to update with the current metadata
         consumer.poll(Duration.ofMillis(0));
+        TestUtils.waitForCondition(() -> client.requests().stream().anyMatch(request -> request.requestBuilder().apiKey().equals(ApiKeys.FIND_COORDINATOR)),
```

Comment:
interesting, makes sense how you keep a single poll, and we just have to wait a bit for the background to actually generate the request. What about encapsulating this bit to make it clearer in all places it's used, to end up with something like:
`TestUtils.waitForCondition(() -> requestGenerated(ApiKeys.FIND_COORDINATOR))`

Reply:
Updated it. Thanks for the suggestion.